src/realm/query.hpp

```diff
@@ -171,6 +171,15 @@ class Query final {
     Query& less_equal(size_t column_ndx, Timestamp value);
     Query& less(size_t column_ndx, Timestamp value);
 
+    // Conditions: size
+    Query& size_equal(size_t column_ndx, int64_t value);
+    Query& size_not_equal(size_t column_ndx, int64_t value);
+    Query& size_greater(size_t column_ndx, int64_t value);
+    Query& size_greater_equal(size_t column_ndx, int64_t value);
+    Query& size_less_equal(size_t column_ndx, int64_t value);
+    Query& size_less(size_t column_ndx, int64_t value);
+    Query& size_between(size_t column_ndx, int64_t from, int64_t to);
```

Comment:
One reason I could think of is the possibility to convert some of the expression based constructs to these older constructs which are faster. This is, however, not done at the moment.